### PR TITLE
Fix add_samples bug

### LINKED ISSUE #111: `add_sample` doesn't perform correctly for padding
the `add_sample` function needs to be fixed, and tests need to be added to confirm that the padding works.

The code:
```
discrete = DiscreteSignal(dt=0.5, samples=np.array([]), carrier_freq=3.0)

t_vals = np.array([0.1, .6, 1.1, 1.6])

discrete3.add_samples(1, [4.0, 3.2])
```

Expected behavior:
`discrete.envelope(t_vals) = array([0.0, 4. +0.j, 3.2+0.j, 0. +0.j])`

Actual behavior: 
`discrete.envelope(t_vals) = array([4. +0.j, 3.2+0.j, 0. +0.j, 0. +0.j])`

### PR DESCRIPTION
**Pull request by DanPuzzuoli**
…g more complicated pulse_to_signals test that also would have caught bug

### Summary

Closes #111

### Details and comments

Fixes bug, adds test directly catching bug, and adds more complicated pulse_to_signals test that would also have caught bug.

### CODE REVIEW

File: test/dynamics/signals/test_pulse_to_signals.py [hunk 1]
```diff
@@ -157,6 +157,78 @@ def test_uneven_pulse_length(self):
         self.assertTrue(signals[0].carrier_freq == 2.0)
         self.assertTrue(signals[1].carrier_freq == 3.0)
 
+    def test_different_start_times(self):
```
Comment (brosand) [inline test/dynamics/signals/test_pulse_to_signals.py:160]: Not really in the scope of this PR, do these pulse schedules use `add_samples`?

File: test/dynamics/signals/test_signals.py [hunk 2]
```diff
@@ -500,12 +500,11 @@ def test_add_samples(self):
...
 
-        t_vals = np.array([0.1, 1.23])
         discrete1.add_samples(0, [4.0, 3.2])
-        self.assertAllClose(discrete1.envelope(t_vals), (4.0, 0.0))
+        self.assertAllClose(discrete1.samples, [4.0, 3.2])
 
-        discrete2.add_samples(3, [1.0, 5.0 + 2j])
-        self.assertAllClose(discrete2.envelope([1.9, 2]), (1.0, 5.0 + 2j))
+        discrete2.add_samples(5, [1.0, 5.0 + 2j])
+        self.assertAllClose(discrete2.samples, [1.0 + 2j, 2.0 + 1j, 3.0, 0.0, 0.0, 1.0, 5.0 + 2j])
```
Comment (brosand) [inline test/dynamics/signals/test_signals.py:507]: One more test for raising an error if samples too short? Never sure when this is overkill or not

**Review by brosand — COMMENTED** [review]
Just three small notes

File: qiskit_dynamics/signals/signals.py [hunk 3]
```diff
@@ -433,35 +433,34 @@ def conjugate(self):
...
-        Appends samples to the pulse starting at start_sample.
-        If start_sample is larger than the number of samples currently
-        in the signal the signal is padded with zeros.
+        Appends samples to the pulse starting at start_sample, filling any gap with zeros.
 
         Args:
-            start_sample: number of the sample at which the new samples
-                should be appended.
-            samples: list of samples to append.
+            start_sample: Number of the sample at which the new samples
```
[the remark this answers is not in the document]
Comment (DanPuzzuoli) [reply qiskit_dynamics/signals/signals.py:439]: Agreed, changing to "index"

File: test/dynamics/signals/test_pulse_to_signals.py (hunk 1 above)
Comment (DanPuzzuoli) [reply test/dynamics/signals/test_pulse_to_signals.py:160]: They do - actually all schedule -> signal conversions do: they are all initialized to empty `DiscreteSignal` instances and the samples for every pulse are appended via `add_samples`. Just a higher level test that depends on this functionality.

**Review by brosand — APPROVED** [review]
LGTM

**Pull request merged**